src/api/proxies/create-dispatch-proxy.js

```diff
@@ -103,14 +103,82 @@ export const createDispatchProxy = (Provider: Client) => {
                 });
               }
 
-              // 9. Parse the JSON from the answer
+              // 9. Did we just delete something?
+              if (callType.type === 'delete') {
+                if (action.pagination) {
+                  // Was it a pagination item? If so, remove its id
+                  // from its related pagination
+                  dispatch({
+                    pagination: {
+                      name: action.pagination.actionName,
+                      id: paginationKey,
+                      entityId: callType.entityId,
+                    },
+                    id: paginationKey,
+                    type: action.pagination.REMOVE,
+                  });
+                  // bail since we don't need to parse the JSON
+                  dispatch({
+                    id: paginationKey,
+                    type: action.SUCCESS,
+                  });
+                } else if (callType.updateEntity) {
+                  const { type, id, updater } = callType.updateEntity;
+
+                  const entity = getState().entities[type][id];
+
+                  dispatch({
+                    type: action.SUCCESS,
+                    entities: {
+                      [type]: {
+                        [id]: updater(entity),
+                      },
+                    },
+                  });
+                }
+
+                return Promise.resolve();
+              }
```

Comment:
No worries about doing it now, we can chat about this much later - but because this single method is getting so long we can spend a little time separating these to separate methods